express/scripts/block-mediator.js

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+const BlockMediator = (() => {
+  const stores = {}; // { [name]: { callbacks: [cb], value: any } }
+
+  const initStore = (name) => {
+    stores[name] = { callbacks: [], value: undefined };
+  };
+
+  const hasStore = (name) => name in stores;
+
+  const listStores = () => Object.keys(stores);
+
+  const get = (name) => stores[name]?.value;
+
+  /**
+   * @param {string} name
+   * @param {any} value
+   * @returns {Promise<{ succeed: boolean, errors: Error[] }>}
+   */
+  const set = (name, value) => {
+    if (!hasStore(name)) {
+      initStore(name);
+    }
+    const oldValue = get(name);
+    stores[name].value = value;
+    return Promise.resolve().then(() => {
```

Comment:
I don't think there's any value in calling `Promise.resolve().then`
You're creating a promise, directly resolving it and then executing the code you actually wanted executed :D
I'd rewrite to something like 


```
const set = (name, value) => {
    if (!hasStore(name)) {
      initStore(name);
    }
    const oldValue = get(name);
    stores[name].value = value;
    const promise = new Promise((resolve) => {
      let success = true;
      const errors = [];
      for (const cb of stores[name].callbacks) {
        try {
          cb({ oldValue, newValue: value });
        } catch (e) {
          success = false;
          errors.push(e);
        }
      }
      resolve({ success, errors });
    });
    return promise;
  };
```

Even then it's a bit strange that success becomes false if 4 callbacks succeed and one fails, but anyway :)